dv_utils/datasets/contract.py

```diff
@@ -0,0 +1,110 @@
+"""
+This module define the data contracts for the dv-utils package.
+"""
+import logging
+import yaml
+
+from ..settings import Settings
+from ..settings import settings as default_settings
+
+from ..connectors.connector import populate_configuration
+from ..connectors import s3,gcs,azure
+
+from datacontract.data_contract import DataContract
+from datacontract.data_contract import Server
+
+logger = logging.getLogger(__name__)
+
+
+class Contract:
+
+    def __init__(self, settings: Settings = None):
+        self.settings: Settings = settings or default_settings
+        self.data_contract=None
+        self.collaboration_space_id=None
+        self.data_descriptor_id=None
+        self.connector=None
+        self.data_connector_config_path=None
+
+    def create_contract(self, data_descriptor: str,collaboration_space_id: str, data_descriptor_id: str,data_connector_config_path:str):
+        #TODO
+        #add version in custom data contract as an attribute in the console
+        #code part with pre-defined dataset
+        #look into filter datasets "echantillonage" to not load all the data https://docs.soda.io/soda-cl/filters.html#in-check-vs-dataset-filters
+        self.collaboration_space_id=collaboration_space_id
+        self.data_descriptor_id=data_descriptor_id
+        self.data_connector_config_path=data_connector_config_path
+        if data_descriptor != None :
+            self.__init_data_connector(data_descriptor)
+            data_contract_yaml=self.__data_descriptor_to_data_contract(data_descriptor)
+            try:
+                logger.info(f"Create data contract object")
+                data_contract = DataContract(data_contract_str=data_contract_yaml)
+                self.data_contract=data_contract
+            except Exception as inst:
+                logger.error(Exception)
```

Comment:
Shouldn't this be `logger.error(inst)` (if we disregard my other comment about using audit_log)?

Reply:
✅ changed (for the audit log, same as before)